### PR TITLE
chore: add types for `destroy`

### LINKED ISSUE #224: Types do not export `destroy` property
The level types do not export a `Level.destroy` property, even though the property does exist on Level. This leads to type errors when using `Level.destroy`.

### PR DESCRIPTION
**Pull request by MicaiahReid**
Fixes #224

### CODE REVIEW

**Comment by vweevers** [comment]
As noted in https://github.com/Level/level/issues/224#issuecomment-1271868543, these types are incompatible with `browser-level`.

That could technically be solved by just adding an overload to the types, but that would wrongly suggest both function signatures are available in node and browsers. What `level` promises to export, is an `abstract-level` implementation. Which is reflected in both its types and the documentation in the README.

We could also document `destroy()` here, but then `level` has to document the differences between `classic-level` and `browser-level` - while its intended purpose is the exact opposite. For convenience, the `level` README does describe some of those differences, but only for individual methods and their options. We have to draw the line somewhere, and its drawn between 1) methods that are part of `abstract-level` and 2) additional implementation-specific methods like `destroy()`.

Bottom line, if you want to use a typed `destroy()`, then replace `level` with `classic-level`. Or import its types, but if you're already directly depending on `classic-level` then `level` serves no use.

**Comment by MicaiahReid** [comment]
I see, thank you for the clarification!
